trackintel/analysis/transport_mode_identification.py

```diff
@@ -110,21 +101,15 @@ def identify_mode(tripleg, wgs, categories):
         str
             the identified mode.
         """
-        # Computes distance over whole tripleg geometry (using the Haversine distance).
-        if wgs:
-            distance = sum([haversine_dist(pt1[0], pt1[1], pt2[0], pt2[1]) for pt1, pt2
-                            in zip(tripleg.geom.coords[:-1], tripleg.geom.coords[1:])])
-        else:
-            distance = tripleg.geom.length
 
         duration = (tripleg['finished_at'] - tripleg['started_at']).total_seconds()
-        speed = distance / duration  # The unit of the speed is m/s
+        speed = tripleg['distance'] / duration  # The unit of the speed is m/s
 
         for bound in categories:
             if speed < bound:
                 return categories[bound]
 
-    triplegs['mode'] = triplegs.apply(lambda l: identify_mode(l, wgs, categories), axis=1)
+    triplegs['mode'] = triplegs.apply(lambda l: identify_mode(l, categories), axis=1)
```

Comment:
This is equivalent to `triplegs.apply(identify_mode, args=(categories), axis=1)`? see [here](https://stackoverflow.com/questions/12182744/python-pandas-apply-a-function-with-arguments-to-a-series)

Reply:
👍